modules/trusted-firmware-m/Kconfig

```diff
@@ -0,0 +1,140 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+config TFM_MINIMAL
+	bool "Use minimal TFM build"
+	imply TFM_IPC
+
+if !TFM_MINIMAL
+source "$(ZEPHYR_BASE)/modules/trusted-firmware-m/Kconfig"
+
+endif
+
+if TFM_MINIMAL
+
+config TFM_MCUBOOT_IMAGE_NUMBER
+	int
+	default 1
+
+# The HASH module is needed to provide RNG functionality
+config TFM_CRYPTO_HASH_MODULE_ENABLED
+	bool
+	default y
+
+config TFM_PARTITION_CRYPTO
+	bool
+	default y
+
+config TFM_PARTITION_PLATFORM
+	bool
+	default y
+
+source "$(ZEPHYR_BASE)/modules/trusted-firmware-m/Kconfig.tfm"
+
+config MBEDTLS_SHA1_C
+	default n
+
+config MBEDTLS_SHA512_C
+	default n
+
+# SHA256 is enabled since we don't override its default
+
+config MBEDTLS_TLS_LIBRARY
+	default n
+
+config MBEDTLS_X509_LIBRARY
+	default n
+
+config MBEDTLS_ENABLE_HEAP
+	default n
+
+config MBEDTLS_PK_WRITE_C
+	default n
+
+config MBEDTLS_RSA_C
+	default n
+
+config MBEDTLS_DHM_C
+	default n
+
+config MBEDTLS_PK_C
+	default n
+
+config MBEDTLS_ECP_C
+	default n
+
+config MBEDTLS_HMAC_DRBG_C
+	default n
+
+config MBEDTLS_CIPHER_MODE_CBC
+	default n
+
+config MBEDTLS_CIPHER_MODE_CTR
+	default n
+
+config MBEDTLS_CMAC_C
+	default n
+
+config MBEDTLS_CCM_C
+	default n
+
+config MBEDTLS_GCM_C
+	default n
+
+config MBEDTLS_CHACHA20_C
+	default n
+
+config MBEDTLS_PK_C
+	default n
+
+config MBEDTLS_POLY1305_C
+	default n
+
+config MBEDTLS_PSA_CRYPTO_STORAGE_C
+	default n
+
+endif # TFM_MINIMAL
+
+if BUILD_WITH_TFM
+
+config PM_PARTITION_SIZE_TFM_SRAM
+	hex
+	prompt "Memory reserved for TFM_RAM." if !TFM_MINIMAL
+	default 0x10000 if TFM_MINIMAL
+	default 0x16000 if SOC_NRF9160
+	default 0x40000 if SOC_NRF5340_CPUAPP
+	help
+	  Memory set aside for the TFM_SRAM partition.
+
+config PM_PARTITION_SIZE_BL2
+	hex "Memory reserved for BL2"
+	default 0x8000 if TFM_BL2
+	default 0
+	help
+	  Memory set aside for the BL2 partition.
+
+
+config PM_PARTITION_SIZE_TFM
+	hex
+	prompt  "Memory reserved for TFM" if !TFM_MINIMAL
+	default 0x8000 if TFM_MINIMAL
```

Comment:
So this partition size must be fixed when using `TFM_MINIMAL`.
Sounds reasonable to me, but the help text should describe this, and why it must be fixed in this case.

Remember that this is rendered into documentation: http://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/kconfig/CONFIG_PM_PARTITION_SIZE_TFM.html

Reply:
good point, I will fix this.